### ISSUE TITLE
Http client request cancelation

### ISSUE THREAD

**Issue by dolphineye** [yhirose/cpp-httplib#86]
Hello,

We currently use your HTTP client in one of our mobile applications. We have a thread pool that sends HTTP requests to fetch data. Yet, we'd sometimes need to cancel a request when necessary as 2G/3G/4G connections usually have latency issues going from 200 msec to more depending on the connection quality.

We considered changing the `Progress` callback's signature so that it returns a boolean value telling whether or not we still want to read the next chunk of data. For example:

```
inline bool read_content_with_length(Stream& strm, std::string& out, size_t len, Progress progress) {
    out.assign(len, 0);
    size_t r = 0;
    while (r < len) {
        auto n = strm.read(&out[r], len - r);
        if (n <= 0) {
            return false;
        }

        r += n;

        if (progress) {
            if (!progress(r, len)) {
                return false;
            }
        }
    }

    return true;
}
```

Yet, I'm not sure it may be quite enough in case the socket is still trying to establish the connection. In the ideal world, when canceling the request, I guess the sooner the better.

Anyway, thank you for all your great work!

**Comment by dolphineye** [comment]
Alright, I've just submitted pull request https://github.com/yhirose/cpp-httplib/pull/89 to introduce this feature. Let me know if you feel that I should bring changes or more testing.

**Comment by yhirose** [comment]
Fixed by #89.
